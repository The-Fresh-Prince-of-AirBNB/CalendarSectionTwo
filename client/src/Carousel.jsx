import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Carousel extends React.Component {
  constructor() {
    super();
    this.state = {
      dates: {
        months: [],
        years: [],
      },
      carousel: [],
    };
    this.generateCal = this.generateCal.bind(this);
    this.changeDates = this.changeDates.bind(this);
  }

  componentDidMount() {
    this.generateCal(new Date().getFullYear(), new Date().getMonth(), 'f');
    this.generateCal(new Date().getFullYear(), new Date().getMonth() + 1, 'f');
  }

  changeDates(direction) {
    const dates = this.state.dates;
    const newCarousel = this.state.carousel;
    if (direction === 'f') {
      newCarousel.shift();
      if (dates.months[1] === 11) {
        this.generateCal(dates.years[1] + 1, 0, direction);
      } else {
        this.generateCal(dates.years[1], dates.months[1] + 1, direction);
      }
      dates.months.shift();
      dates.years.shift();
    } else {
      newCarousel.pop();
      if (dates.months[1] === 0) {
        this.generateCal(dates.years[1] - 1, 11, direction);
      } else {
        this.generateCal(dates.years[1], dates.months[1] - 1, direction);
      }
      dates.months.pop();
      dates.years.pop();
    }
    this.setState({
      dates: dates,
      carousel: newCarousel,
    });
  }

  generateCal(y, m, dir) {
    // Create an array of the empty days on the first week of the month
    const emptyDays = [];
    for (let i = 0; i < (new Date(y, m)).getDay(); i += 1) {
      emptyDays.push(<td className="day">{''}</td>);
    }
    // Create an array of all the other days in the month
    const daysInTheMonth = [];
    for (let i = 1; i <= (new Date(y, m, 0).getDate()); i += 1) {
      daysInTheMonth.push(<td className="day calDay">{i}</td>);
    }
    // Concat the two together to get an array showing all empty days and days in the month
    const total = emptyDays.concat(daysInTheMonth);
    // Define a rows array to add your 'row' of 7 <td> cells elements into
    const rows = [];
    let row = [];
    // Iterate through the complete array
    for (let i = 0; i < total.length; i += 1) {
      // If the day is not divisible by 7, add it to the row
      if (i % 7 !== 0) {
        row.push(total[i]);
      // Otherwise, push the entire row into the rows array,
      // empty out the row, and add the next cell into the row
      } else {
        rows.push(row);
        row = [];
        row.push(total[i]);
      }
      // At the end, push the last row into the rows array
      if (i === total.length - 1) {
        rows.push(row);
      }
    }
    // Create the table for the entire month
    const calendar = rows.map((r) => <tr>{r}</tr>);
    const newCarousel = this.state.carousel;
    newCarousel.push(calendar);
    const newDates = this.state.dates;
    if (dir === 'f') {
      newDates.months.push(m);
      newDates.years.push(y);
    } else {
      newDates.months.unshift(m);
      newDates.years.unshift(y);
    }
    this.setState({
      dates: newDates,
      carousel: newCarousel,
    });
  }

  render() {
    return (
      <div>
        <button onClick={() => this.changeDates('b')} type="button">
          Before
        </button>
        <button onClick={() => this.changeDates('f')} type="button">
          Next
        </button>
        <div className="carousel">
          <div>
            <div>November, 2020</div>
            <table className="table">
              <thead>
                <tr>
                  <th>Su</th>
                  <th>Mo</th>
                  <th>Tu</th>
                  <th>We</th>
                  <th>Th</th>
                  <th>Fr</th>
                  <th>Sa</th>
                </tr>
              </thead>
              <tbody>
                {this.state.carousel[0]}
              </tbody>
            </table>
          </div>
          <div>
            <div>December, 2020</div>
            <table className="table">
              <thead>
                <tr>
                  <th>Su</th>
                  <th>Mo</th>
                  <th>Tu</th>
                  <th>We</th>
                  <th>Th</th>
                  <th>Fr</th>
                  <th>Sa</th>
                </tr>
              </thead>
              <tbody>
                {this.state.carousel[1]}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    );
  }
};

export default Carousel;
