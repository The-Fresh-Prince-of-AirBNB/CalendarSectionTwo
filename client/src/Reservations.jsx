/* eslint-disable import/extensions */
import React from 'react';
import Box from './Box.jsx';
import styles from '../../styles.css';

const Reservations = () => (
  <div>
    <img className={styles.header} src="../dist/photos/airbnb_header.png" alt="header" />
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
      <div style={{ width: '52%' }}>
        <img className={styles.airbnbsummary} src="../dist/photos/airbnb_summary.png" alt="summary" />
        <img className={styles.description} src="../dist/photos/airbnb_description.png" alt="description" />
        <img className={styles.description} src="../dist/photos/airbnb_amenities.png" alt="amenities" />
      </div>
      <Box />
    </div>
  </div>
);

export default Reservations;
