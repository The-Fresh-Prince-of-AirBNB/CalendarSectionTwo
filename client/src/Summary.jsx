import React, { useState } from 'react';

// eslint-disable-next-line arrow-body-style
const Summary = (props) => {
  const lineStyle = { textDecoration: 'underline' };

  return (
    <div>
      <button
        className="checkAvailabilityAndReserve"
        type="button"
        style={{ backgroundPosition: `${props.x - 80}px ${props.y}px`, backgroundImage: 'linear-gradient(to right, #d43974, #d43939, #d43974)' }}
        onMouseMove={(event) => props.mousePosition(event)}
        onMouseLeave={() => {
          props.setX(0);
          props.setY(0);
        }}
      >
        Reserve
      </button>
      <div style={{ fontSize: '13px', textAlign: 'center', margin: '20px 0px' }}>You won&apos;t be charged yet</div>
      <div className="summaryItems">
        <div className="summaryItem">
          <div style={lineStyle}>
            $
            {props.fees.nightlyFee}
            {' '}
            x 4 nights
          </div>
        </div>
        <div className="summaryItem">
          <div style={lineStyle}>Cleaning fee</div>
          <div>
            $
            {props.fees.cleaningFee}
          </div>
        </div>
        <div className="summaryItem">
          <div style={lineStyle}>Service fee</div>
          <div>
            $
            {props.fees.serviceFee}
          </div>
        </div>
        <div className="summaryItem">
          <div style={lineStyle}>Occupancy taxes and fees</div>
          <div>
            $
            {props.fees.taxes}
          </div>
        </div>
      </div>
      <div className="total">
        <div>Total</div>
        <div>$ x</div>
      </div>
    </div>
  );
};

export default Summary;
