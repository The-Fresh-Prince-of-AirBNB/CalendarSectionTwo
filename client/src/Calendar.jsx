import React, { useState, useEffect, useRef } from 'react';
import Carousel from './Carousel.jsx';

const Calendar = (props) => {
  const checkInRef = useRef(null);
  const [placeholders, setPlaceholders] = useState({ checkIn: 'Add date', checkOut: 'Add date' });
  const [book, setBook] = useState({ start: '', end: '' });
  const [box, setBox] = useState('checkIn');

  const toggleCheckin = (event) => {
    if (book.start === '') {
      if (checkInRef.current && !checkInRef.current.contains(event.target)) {
        setPlaceholders({ checkIn: 'Add date', checkOut: placeholders.checkOut });
      } else {
        setPlaceholders({ checkIn: 'MM/DD/YYY', checkOut: placeholders.checkOut });
      }
    } else if (checkInRef.current && !checkInRef.current.contains(event.target)) {
      setPlaceholders({ checkIn: placeholders.checkIn, checkOut: 'Add date' });
    } else {
      setPlaceholders({ checkIn: placeholders.checkIn, checkOut: 'MM/DD/YYY' });
    }
  };

  const handleBook = (m, d, y, out) => {
    if (m && !out) {
      setBook({ start: `${m + 1}/${d}/${y}`, end: '' });
      props.setForm({ in: `${m + 1}/${d}/${y}`, out: 'Add date' });
      setBox('checkOut');
    } else if (out) {
      setBook({ start: book.start, end: `${m + 1}/${d}/${y}` });
      props.setForm({ in: props.form.in, out: `${m + 1}/${d}/${y}` });
      props.setReserve(!props.reserve);
    } else {
      setBook({ start: '', end: '' });
      props.setForm({ in: 'Add date', out: 'Add date' });
      setBox('checkIn');
      props.setReserve(false);
    }
  };

  const handleChange = (e, type) => {
    e.preventDefault();
    if (type === 's') {
      setBook({ start: e.target.value, end: book.end });
    } else {
      setBook({ start: book.start, end: e.target.value });
    }
  };

  useEffect(() => {
    document.addEventListener('click', toggleCheckin, false);
    return () => {
      document.removeEventListener('click', toggleCheckin, false);
    };
  });

  let checkInBox;
  let checkOutBox;
  if (box === 'checkOut') {
    checkOutBox = (
      <div style={{ border: '2px solid black', borderRadius: '8px' }} className="calCheckOut" ref={checkInRef}>
        <div>CHECK-OUT</div>
        <input type="text" className="calCheckInput" style={{ border: 'none', width: '70%' }} placeholder={placeholders.checkOut} onClick={() => toggleCheckin} value={book.end} onChange={(e) => handleChange(e, 'e')} />
      </div>
    );
    checkInBox = (
      <div className="calCheckIn" ref={checkInRef}>
        <div>CHECK-IN</div>
        <input type="text" className="calCheckInput" style={{ border: 'none', width: '70%' }} placeholder={placeholders.checkIn} onClick={() => toggleCheckin} value={book.start} onChange={(e) => handleChange(e, 's')} />
      </div>
    );
  } else {
    checkOutBox = (
      <div className="calCheckOut mask">
        <div>CHECK-OUT</div>
        <div>Add date</div>
      </div>
    );
    checkInBox = (
      <div style={{ border: '2px solid black', borderRadius: '8px' }} className="calCheckIn" ref={checkInRef}>
        <div>CHECK-IN</div>
        <input type="text" className="calCheckInput" style={{ border: 'none', width: '70%' }} placeholder={placeholders.checkIn} onClick={() => toggleCheckin} value={book.start} onChange={(e) => handleChange(e, 's')} />
      </div>
    );
  }

  return (
    <div className="calendar">
      <div className="calendarContent">
        <div>
          <div style={{ fontSize: '22px' }}>Select Dates</div>
          <div>
            Minimum stay:
            {' '}
            {props.fees.minNights}
          </div>
        </div>
        <form className="calForm">
          {checkInBox}
          {checkOutBox}
        </form>
        <div className="calCarousel">
          <Carousel
            setFees={props.setFees}
            setDates={props.setDates}
            dates={props.dates}
            fees={props.fees}
            close={props.close}
            handleBook={handleBook}
          />
        </div>
      </div>
    </div>
  );
};

export default Calendar;
