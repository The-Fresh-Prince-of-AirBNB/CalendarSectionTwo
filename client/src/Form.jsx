/* eslint-disable import/extensions */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Calendar from './Calendar.jsx';
import Guests from './Guests.jsx';
import Summary from './Summary.jsx';

const Form = ({
  fees, dates, setFees, setDates,
}) => {
  const [displayCalendar, setDisplayCalendar] = useState(false);
  const [displayGuests, setDisplayGuests] = useState(false);
  const [form, setForm] = useState({ in: 'Add date', out: 'Add date', days: 0 });
  const [guestBorder, setGuestBorder] = useState(false);
  const [x, setX] = useState(0);
  const [y, setY] = useState(0);
  const [reserve, setReserve] = useState(false);
  const [guests, setGuest] = useState({ adults: 2, children: 0, infants: 0 });
  const [hovering, setHover] = useState(false);
  const max = guests.adults + guests.children;
  const styling = {
    hover: { backgroundPosition: `${x + 60}px ${y}px`, backgroundImage: 'linear-gradient(to right, #d93275, #d93243, #d93275)' },
    static: { backgroundImage: 'linear-gradient(to right, #d93243, #d93275, #d93275)' },
  };

  const add = {
    adults: (q) => {
      if ((q === -1 && guests.adults > 0) || (q === 1 && max < 4)) {
        setGuest({ adults: guests.adults + q, children: guests.children, infants: guests.infants });
      }
    },
    chilren: (q) => {
      if ((q === -1 && guests.children > 0) || (q === 1 && max < 4)) {
        setGuest({ adults: guests.adults, children: guests.children + q, infants: guests.infants });
      }
    },
    infants: (q) => {
      if ((q === -1 && guests.infants > 0) || q === 1) {
        setGuest({ adults: guests.adults, children: guests.children, infants: guests.infants + q });
      }
    },
  };

  const toggleCalendar = () => (
    setDisplayCalendar(!displayCalendar)
  );

  const toggleGuests = () => {
    setGuestBorder(!guestBorder);
    setDisplayGuests(!displayGuests);
  };

  const closeGuests = () => {
    setGuestBorder(false);
    setDisplayGuests(false);
  };

  const mousePosition = (event) => {
    setX(event.clientX);
    setY(event.clientY);
  };

  return (
    <div>
      <div className="form">
        <div>
          <button
            type="button"
            className="check"
            style={{ borderBottom: guestBorder ? 'none' : '1px solid rgb(179, 179, 179)' }}
            onClick={() => {
              toggleCalendar();
              closeGuests();
            }}
          >
            <div className="checkIn">
              <div style={{ fontSize: '10px', marginBottom: '5px' }}>CHECK-IN</div>
              <div style={{ color: form.in === 'Add date' ? 'rgb(125, 125, 125)' : 'black', fontFamily: 'Montserrat, sans-serif' }}>{form.in}</div>
            </div>
            <div className="checkOut">
              <div style={{ fontSize: '10px', marginBottom: '5px' }}>CHECKOUT</div>
              <div style={{ color: form.out === 'Add date' ? 'rgb(125, 125, 125)' : 'black', fontFamily: 'Montserrat, sans-serif' }}>{form.out}</div>
            </div>
          </button>
        </div>
        <div role="button" tabIndex={0} style={{ display: displayCalendar ? 'block' : 'none' }} onKeyDown={() => setDisplayGuests(false)}>
          <Calendar
            close={toggleCalendar}
            setFees={setFees}
            setDates={setDates}
            dates={dates}
            fees={fees}
            setForm={setForm}
            form={form}
            setReserve={setReserve}
            reserve={reserve}
          />
        </div>
        <div>
          <button
            type="button"
            className="guestBut"
            style={{ border: guestBorder ? '2px solid black' : 'none', borderRadius: '8px' }}
            onClick={() => {
              toggleGuests();
              setGuestBorder(!guestBorder);
            }}
          >
            <div style={{ fontSize: '10px', marginBottom: '5px' }}>GUESTS</div>
            <div style={{ fontFamily: 'Montserrat, sans-serif' }}>
              {max}
              {' '}
              guests
            </div>
          </button>
          <div style={{ display: displayGuests ? 'block' : 'none' }}>
            <Guests
              toggleGuests={toggleGuests}
              guests={guests}
              max={max}
              add={add}
            />
          </div>
        </div>
      </div>
      {reserve
        ? (
          <Summary
            fees={fees}
            max={max}
            x={x}
            y={y}
            setX={setX}
            setY={setY}
            mousePosition={mousePosition}
            form={form}
            setHover={setHover}
            hovering={hovering}
            styling={styling}
          />
        )
        : (
          <div>
            <button
              type="button"
              className="checkAvailabilityAndReserve"
              style={hovering ? styling.hover : styling.static}
              onMouseMove={(event) => {
                mousePosition(event);
                setHover(true);
              }}
              onMouseLeave={() => {
                setHover(false);
                setX(0);
                setY(0);
              }}
              onClick={() => toggleCalendar()}
            >
              <div style={{ display: 'inline' }}>
                Check Availability
              </div>
            </button>
          </div>
        )}
    </div>
  );
};

Form.propTypes = {
  setFees: PropTypes.func.isRequired,
  setDates: PropTypes.func.isRequired,
  fees: PropTypes.shape.isRequired,
  dates: PropTypes.shape.isRequired,
};

export default Form;
