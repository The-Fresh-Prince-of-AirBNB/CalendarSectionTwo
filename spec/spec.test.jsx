/* eslint-disable no-unused-vars */
/* eslint-disable import/extensions */
/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import {
  configure, shallow, mount,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'core-js/stable';
import 'regenerator-runtime/runtime';

// Import all components for testing
import Guests from '../client/src/Guests.jsx';
import Form from '../client/src/Form.jsx';
import Carousel from '../client/src/Carousel.jsx';
import Calendar from '../client/src/Calendar.jsx';
import Reservations from '../client/src/Reservations.jsx';
import Summary from '../client/src/Summary.jsx';

configure({ adapter: new Adapter() });
afterEach(cleanup);

describe('testing for jest', () => {
  it('test to see that jest runs', () => {
    const data = { one: 1 };
    data.two = 2;
    expect(data).toEqual({ one: 1, two: 2 });
    expect(1 + 2).toBe(3);
    const n = null;
    expect(n).toBeNull();
  });
});

// Sections for all components
describe('Reservations', () => {
  it('should render Reservations without crashing', () => {
    const res = shallow(<Reservations />);
    expect(res).toMatchSnapshot();
  });
});

describe('Form', () => {
  const date = {
    start: [2020, 11, 20],
    end: [2020, 11, 25],
  };

  it('should render Form without crashing', () => {
    const fo = shallow(<Form />);
    expect(fo).toMatchSnapshot();
  });

  it('accepts date props', () => {
    const wrapper = mount(<Form date={date} />);
    expect(wrapper.props().date).toEqual(date);
  });

  it('opens up calendar when calendar button is clicked on form', () => {
    const wrapper = mount(<Form />);
    const calButton = wrapper.find({ 'data-testid': 'formCalendarTest' });
    let cal = wrapper.find({ 'data-testid': 'openCalendarTest' });
    expect(cal.prop('style')).toHaveProperty('display', 'none');
    calButton.simulate('click');
    wrapper.update();

    cal = wrapper.find({ 'data-testid': 'openCalendarTest' });
    expect(cal.prop('style')).toHaveProperty('display', 'block');
  });

  it('closes the guest list when calendar button is clicked on form', () => {
    const wrapper = mount(<Form />);
    const calButton = wrapper.find({ 'data-testid': 'formCalendarTest' });
    const guestButton = wrapper.find({ 'data-testid': 'buttonGuestTest' });
    let guestList = wrapper.find({ 'data-testid': 'closeGuestTest' });
    // open up the guest list, expect style from none to block
    expect(guestList.prop('style')).toHaveProperty('display', 'none');
    guestButton.simulate('click');
    wrapper.update();
    guestList = wrapper.find({ 'data-testid': 'closeGuestTest' });
    expect(guestList.prop('style')).toHaveProperty('display', 'block');
    // open up the calendar, want the guest list to close when calendar opens
    calButton.simulate('click');
    wrapper.update();
    guestList = wrapper.find({ 'data-testid': 'closeGuestTest' });
    const cal = wrapper.find({ 'data-testid': 'openCalendarTest' });
    expect(guestList.prop('style')).toHaveProperty('display', 'none');
    expect(cal.prop('style')).toHaveProperty('display', 'block');
  });
});

describe('Guests', () => {
  const guests = { adults: 2, children: 2, infants: 0 };

  it('should render Guests without crashing', () => {
    const gu = shallow(<Guests />);
    expect(gu).toMatchSnapshot();
  });

  it('accepts guests props', () => {
    const wrapper = mount(<Guests guests={guests} />);
    expect(wrapper.props().guests.adults).toEqual(2);
    expect(wrapper.props().guests.adults).not.toEqual(3);
  });

  it('adds no more than 4 guests', async () => {
    const formWrapper = mount(<Form />);
    const guestWrapper = mount(<Guests />);

    let totAdults = guestWrapper.find({ 'data-testid': 'totAdults' });
    const addAdultsBut = guestWrapper.find({ 'data-testid': 'addAdults' });
    const addChildrenBut = guestWrapper.find({ 'data-testid': 'addChildren' });
    // Start off with 2 guests
    expect(totAdults.text()).toBe('2');
    // Simulate two clicks on the add buttons
    addAdultsBut.simulate('click');
    addAdultsBut.simulate('click');
    guestWrapper.update();
    formWrapper.update();

    totAdults = guestWrapper.find({ 'data-testid': 'totAdults' });
    expect(totAdults.text()).toBe('4');

    addAdultsBut.simulate('click');
    guestWrapper.update();
    totAdults = guestWrapper.find({ 'data-testid': 'totAdults' });
    expect(totAdults.text()).toBe('4');
  });

  it('calls a function in Form that renders the form', () => {
    const mock = jest.fn();
    const wrapper = shallow(<Guests setGuest={mock} />);
    const addAdultBut = wrapper.find({ 'data-testid': 'addAdults' });

    addAdultBut.simulate('click');
    expect(mock).toHaveBeenCalled();
  });
});

describe('Calendar', () => {
  const reserve = false;

  it('should render Calendar without crashing', () => {
    const ca = shallow(<Calendar />);
    expect(ca).toMatchSnapshot();
  });

  it('accepts reserve props', () => {
    const wrapper = mount(<Calendar reserve={reserve} />);
    expect(wrapper.props().reserve).toBeFalsy();
  });
});

describe('Carousel', () => {
  const testFunc = (x, y) => x + y;

  it('should render Carousel without crashing', () => {
    const car = shallow(<Carousel />);
    expect(car).toMatchSnapshot();
  });

  it('accepts function props', () => {
    const wrapper = mount(<Carousel testFunc={testFunc} />);
    expect(wrapper.props().testFunc(2, 3)).toEqual(5);
  });

  it('Calls the API call function on page load', () => {
    const spy = jest.spyOn(Carousel.prototype, 'getExisting');
    const wrapper = mount(<Carousel />);
    wrapper.instance().getExisting();
    expect(spy).toHaveBeenCalled();

    afterEach(() => {
      spy.mockClear();
    });
  });
});

describe('Summary', () => {
  const form = { in: '11/23/2020', out: '11/27/2020', days: 4 };
  const fees = {
    nightlyFee: 150,
    cleaningFee: 30,
    serviceFee: 20,
    taxes: 15,
    minNights: 4,
  };

  it('should render Summary without crashing', () => {
    const su = shallow(<Summary />);
    expect(su).toMatchSnapshot();
  });

  it('accepts a form (object) props', () => {
    const wrapper = mount(<Summary form={form} />);
    expect(wrapper.props().form.in).toEqual('11/23/2020');
  });
});
