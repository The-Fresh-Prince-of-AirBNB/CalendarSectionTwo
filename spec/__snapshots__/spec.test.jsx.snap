// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render Calendar without crashing 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "fontSize": "22px",
          }
        }
      >
        Select Dates
      </div>
      <div
        style={
          Object {
            "color": "rgb(160, 160, 160)",
            "marginTop": "8px",
          }
        }
      >
        Minimum stay:
         
        0
         
        nights
      </div>
    </div>
    <form>
      <div
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "8px",
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <div
          style={
            Object {
              "paddingLeft": "4px",
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": "10px",
                "marginBottom": "3px",
              }
            }
          >
            CHECK-IN
          </div>
          <input
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            placeholder="Add date"
            style={
              Object {
                "border": "none",
                "fontFamily": "Montserrat, sans-serif",
                "marginLeft": "-2px",
                "width": "70%",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="undefined undefined"
      >
        <div
          style={
            Object {
              "fontSize": "10px",
              "marginBottom": "3px",
            }
          }
        >
          CHECKOUT
        </div>
        <div
          style={
            Object {
              "fontSize": "13px",
              "lineHeight": "14px",
            }
          }
        >
          Add date
        </div>
      </div>
    </form>
    <div>
      <Carousel
        close={[Function]}
        dates={
          Object {
            "reservations": Object {},
          }
        }
        fees={
          Object {
            "cleaningFee": 0,
            "minNights": 0,
            "nightlyFee": 0,
            "serviceFee": 0,
            "taxes": 0,
          }
        }
        handleBook={[Function]}
        setDates={[Function]}
        setFees={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Carousel should render Carousel without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "marginRight": "4.5%",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "height": "32px",
            "width": "32px",
          }
        }
      >
         
      </div>
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginBottom": "5px",
            "marginRight": "30%",
          }
        }
      >
        January
         
        0
      </div>
    </div>
    <table>
      <thead>
        <tr
          style={
            Object {
              "color": "rgb(129, 129, 129)",
            }
          }
        >
          <th>
            Su
          </th>
          <th>
            Mo
          </th>
          <th>
            Tu
          </th>
          <th>
            We
          </th>
          <th>
            Th
          </th>
          <th>
            Fr
          </th>
          <th>
            Sa
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": "18px",
            "marginBottom": "5px",
            "marginLeft": "30%",
          }
        }
      >
        February
         
        0
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
    </div>
    <table>
      <thead>
        <tr
          style={
            Object {
              "color": "rgb(129, 129, 129)",
            }
          }
        >
          <th>
            Su
          </th>
          <th>
            Mo
          </th>
          <th>
            Tu
          </th>
          <th>
            We
          </th>
          <th>
            Th
          </th>
          <th>
            Fr
          </th>
          <th>
            Sa
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Clear dates
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`Form should render Form without crashing 1`] = `
<div>
  <div>
    <div>
      <button
        data-testid="formCalendarTest"
        onClick={[Function]}
        style={
          Object {
            "borderBottom": "1px solid rgb(179, 179, 179)",
          }
        }
        type="button"
      >
        <div>
          <div
            style={
              Object {
                "fontSize": "10px",
                "marginBottom": "5px",
              }
            }
          >
            CHECK-IN
          </div>
          <div
            style={
              Object {
                "color": "rgb(125, 125, 125)",
                "fontFamily": "Montserrat, sans-serif",
              }
            }
          >
            Add date
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "fontSize": "10px",
                "marginBottom": "5px",
              }
            }
          >
            CHECKOUT
          </div>
          <div
            style={
              Object {
                "color": "rgb(125, 125, 125)",
                "fontFamily": "Montserrat, sans-serif",
              }
            }
          >
            Add date
          </div>
        </div>
      </button>
    </div>
    <div
      data-testid="openCalendarTest"
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={0}
    >
      <Calendar
        close={[Function]}
        dates={
          Object {
            "reservations": Object {},
          }
        }
        fees={
          Object {
            "cleaningFee": 0,
            "minNights": 0,
            "nightlyFee": 0,
            "serviceFee": 0,
            "taxes": 0,
          }
        }
        form={
          Object {
            "days": 0,
            "in": "Add date",
            "out": "Add date",
          }
        }
        reserve={false}
        setDates={[Function]}
        setFees={[Function]}
        setForm={[Function]}
        setReserve={[Function]}
      />
    </div>
    <div>
      <button
        data-testid="buttonGuestTest"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "border": "none",
            "borderRadius": "8px",
            "display": "flex",
            "justifyContent": "space-between",
            "padding": "10px 15px",
          }
        }
        type="button"
      >
        <div>
          <div
            style={
              Object {
                "fontSize": "10px",
                "marginBottom": "5px",
              }
            }
          >
            GUESTS
          </div>
          <div
            data-testid="numOfGuests"
            style={
              Object {
                "fontFamily": "Montserrat, sans-serif",
              }
            }
          >
            2
             
            guests
          </div>
        </div>
        <div
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "lighter",
            }
          }
        >
          v
        </div>
      </button>
      <div
        data-testid="closeGuestTest"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <Guests
          guests={
            Object {
              "adults": 2,
              "children": 0,
              "infants": 0,
            }
          }
          setGuest={[Function]}
          toggleGuests={[Function]}
        />
      </div>
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      style={
        Object {
          "backgroundImage": "linear-gradient(to right, #d93243, #d93275, #d93275)",
        }
      }
      type="button"
    >
      <div
        style={
          Object {
            "display": "inline",
          }
        }
      >
        Check Availability
      </div>
    </button>
  </div>
</div>
`;

exports[`Guests should render Guests without crashing 1`] = `
<div>
  <div>
    <div>
      <div>
        <div>
          Adults
        </div>
        <div>
          <button
            className="undefined undefined"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <div
            data-testid="totAdults"
          >
            2
          </div>
          <button
            className="undefined undefined"
            data-testid="addAdults"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <div>
        <div>
          <div>
            Children
          </div>
          <div
            style={
              Object {
                "fontSize": "12px",
                "marginTop": "5px",
              }
            }
          >
            Ages 2–12
          </div>
        </div>
        <div>
          <button
            className="undefined undefined"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <div>
            0
          </div>
          <button
            className="undefined undefined"
            data-testid="addChildren"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <div>
        <div>
          <div>
            Infants
          </div>
          <div
            style={
              Object {
                "fontSize": "12px",
                "marginTop": "5px",
              }
            }
          >
            Under 2
          </div>
        </div>
        <div>
          <button
            className="undefined undefined"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <div>
            0
          </div>
          <button
            className="undefined undefined"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "fontSize": "13px",
            "lineHeight": "20px",
          }
        }
      >
        4 guests maximum. Infants don’t count toward the number of guests.
      </div>
    </div>
    <button
      className="undefined undefined"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`Reservations should render Reservations without crashing 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "display": "inline",
            "fontSize": "24px",
            "fontWeight": "500",
          }
        }
      >
        $
         
        0
      </div>
      <div
        style={
          Object {
            "display": "inline",
          }
        }
      >
         / night
      </div>
    </div>
    <Form
      dates={
        Object {
          "reservations": Object {},
        }
      }
      fees={
        Object {
          "cleaningFee": 0,
          "minNights": 0,
          "nightlyFee": 0,
          "serviceFee": 0,
          "taxes": 0,
        }
      }
      setDates={[Function]}
      setFees={[Function]}
    />
  </div>
</div>
`;

exports[`Summary should render Summary without crashing 1`] = `
<div>
  <button
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={Object {}}
    type="button"
  >
    Reserve
  </button>
  <div
    style={
      Object {
        "fontSize": "13px",
        "margin": "20px 0px",
        "textAlign": "center",
      }
    }
  >
    You won't be charged yet
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        $
        0
         
        x
         
        0
         
        nights
      </div>
      <div>
        $
        0
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Cleaning fee
      </div>
      <div>
        $
        0
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Service fee
      </div>
      <div>
        $
        30
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Occupancy taxes and fees
      </div>
      <div>
        $
        20
      </div>
    </div>
  </div>
  <div>
    <div>
      Total
    </div>
    <div>
      $
      50
    </div>
  </div>
</div>
`;
